admin-dev/themes/new-theme/js/app/components/CheckboxesDropdown.vue

```diff
@@ -72,65 +73,80 @@
 
 <script lang="ts">
   import {defineComponent, PropType} from 'vue';
-  import ProductEventMap from '@pages/product/product-event-map';
-
-  const CombinationEvents = ProductEventMap.combinations;
+  import EventEmitter from '@components/event-emitter';
 
   export default defineComponent({
-    name: 'FilterDropdown',
-    data(): {selectedFilters: Array<Record<string, any>>} {
+    data(): {
+      selectedItems: Array<Record<string, any>>
+    } {
       return {
-        selectedFilters: [],
+        selectedItems: [],
       };
     },
     props: {
       parentId: {
         type: Number,
-        required: true,
+        default: 1,
       },
-      children: {
+      items: {
         type: Array as PropType<Array<Record<string, any>>>,
         required: true,
       },
+      initialItemIds: {
+        type: Array as PropType<any>,
+        default: () => [],
+      },
       label: {
         type: String,
         required: true,
       },
       eventEmitter: {
-        type: Object,
+        type: Object as PropType<typeof EventEmitter>,
         required: true,
       },
+      // provide this property if you need to clear all selected events from parent component
+      clearSelectedItemsEvent: {
+        type: String,
+        default: '',
+      },
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
     },
     mounted() {
-      this.eventEmitter.on(CombinationEvents.clearAllCombinationFilters, this.clear);
+      this.selectedItems = this.items.filter((item) => this.initialItemIds.includes(item.id));
+      if (this.clearSelectedItemsEvent) {
```

Comment:
I think we could get rid of this event thing, the idea is to be able to clear the selected items from the parent. Now your `selectedItems` is a local data so it can't be edited by the parent, however it's built on mount based on `initialItemIds` And this one is a prop that can be edited by the parent. SO hat is possible is:

1. turn `selectedItems` into a computed field, based on `this.initialItemIds`
```ts
selectedItems(): array {
  this.items.filter((item) => this.initialItemIds.includes(item.id))
}
```

Now it will automatically update when the prop is updated, so in the parent you can do `mycomponent.initialItemIds = []` and it should clear it (it should also work ith automatic binding if the prop is linked to another variable).

Maybe you'll have to update also the `toggleSelection` method, basically instead of updating `selectedItems` (which is now a computed field without a setter) you do the same operations on `initialItemIds` and it will automatically update `selectedItems`

And finally maybe `initialItemIds` can be renamed into `selectedItemIds` which would match more with the way it will work (easier to understand from the parent point of view).

Reply:
With this you should be able to remove the whole event emitter thing which would make the component more stable and more reusable

Reply:
Oh and with this maybe you also won't need to use the `removeItem` and `addItem` event, but a `change` event so that the parent can use `selectedItemIds` directly as v-model

Reply:
all of it sounds great in theory, but you cannot edit props. All props that are passed from parent are kind of immutable as I understand.
<img width="782" alt="Screenshot 2023-02-03 at 14 54 12" src="https://user-images.githubusercontent.com/31609858/216609869-b44faf33-650a-4ebc-9fb0-e1ed8105d501.png">

Reply:
oh I was wrong all the time about this, since I passed the selected ids from parent, then child doesn't need to update them anymore. Fixed in latest commits.